core/src/com/unciv/models/Spy.kt

```diff
@@ -115,19 +116,12 @@ class Spy private constructor() : IsPartOfGameInfoSerialization {
                 setAction(SpyAction.StealingTech) // There are new techs to steal!
             }
             SpyAction.StealingTech -> {
-                val stealableTechs = espionageManager.getTechsToSteal(getCity().civ)
-                if (stealableTechs.isEmpty()) {
+                turnsRemainingForAction = getTurnsRemainingToStealTech()
```

Comment:
Would removing hasCountdownTurns be fine? It's a little weird how only some enum types use it.
Otherwise it might be better if we also have getTurnsRemaining() which includes getTurnsRemainingToStealTech(), city-state elections and counter intelligence.